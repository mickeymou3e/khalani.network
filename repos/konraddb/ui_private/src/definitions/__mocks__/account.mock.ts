export const customerCodeMock = "26a17540-2bf0-428d-b902-33f035bcbeb7";

export const clientMock = {
  address: "B",
  bfid: "5317253",
  city: "beEe",
  code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  company_name: "Company1",
  contact_person_name: "Company1",
  country: "AqE",
  created_at: "2023-05-10T08:07:39Z",
  currency: "EUR",
  deposit_holder_code: "4b99350c-70f5-4d20-ac45-22d11709f4a9",
  depot: "877",
  domain: "3911",
  email: "ndhur@mail.com",
  fee: "4",
  fix_default_customer_code: null,
  fix_is_enabled: false,
  fix_reset_seq_number_on_logon: false,
  fix_sender_comp_id: null,
  ibans: [
    {
      currency: "EUR",
      iban: "DE11416500013918529410",
      is_enabled: true,
    },
  ],
  is_enabled: true,
  kontokorrent: "281",
  phone_number: "34552644",
  rubric: "475",
  vat: "644",
  zip_code: "848",
};

export const userProfileMock = {
  client: clientMock,
  code: "0680f2b1-df2e-48ce-a988-7c7a2d37d8f6",
  customer_code: customerCodeMock,
  display_name: "steve@mail.com",
  features: [],
  name: "steve",
  registration_date: "2023-06-21T12:33:19Z",
  role: "mastertrader",
};
