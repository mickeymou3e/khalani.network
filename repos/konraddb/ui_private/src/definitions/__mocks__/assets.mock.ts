export const assignedAssetsMock = [
  {
    base: "BTC",
    created: "0001-01-01T00:00:00.000Z",
    guaranteed_qty: "1",
    quote: "EUR",
    pair: "BTC/EUR",
    settings_code: "d529e121-8a3b-461d-967f-a13f9c1f5660",
    stream_type: "standard",
    updated: "0001-01-01T00:00:00.000Z",
  },
  {
    base: "ETH",
    created: "0001-01-01T00:00:00.000Z",
    guaranteed_qty: "2.5",
    quote: "EUR",
    pair: "ETH/EUR",
    settings_code: "0b277b15-04ea-42e0-b330-668c0d71a71c",
    stream_type: "standard",
    updated: "0001-01-01T00:00:00.000Z",
  },
  {
    base: "ATOM",
    created: "0001-01-01T00:00:00.000Z",
    guaranteed_qty: "100",
    quote: "EUR",
    pair: "ATOM/EUR",
    settings_code: "ebc004a5-e182-45e3-a4b1-8a8c103b8797",
    stream_type: "standard",
    updated: "0001-01-01T00:00:00.000Z",
  },
  {
    base: "JLT-F23",
    created: "2023-10-30T15:36:44.651Z",
    guaranteed_qty: "10000000",
    quote: "EUR",
    pair: "JLT-F23/EUR",
    settings_code: "c33a177c-bde7-42f5-b72f-5bd36b537311",
    spread_mode: "strict",
    stream_type: "standard",
    updated: "2023-10-30T15:36:41.489Z",
  },
];

export const BTCAssetMock = { label: "BTC", description: "Bitcoin" };
export const BTCUSDAssetMock = {
  icon: "BTC",
  label: "BTC/USD",
  description: "Bitcoin",
};
export const ETHAssetMock = { label: "ETH", description: "Ethereum" };
export const USDCAssetMock = { label: "USDC", description: "USD Coin" };
export const XRPAssetMock = { label: "XRP", description: "XRP" };
export const USDAssetMock = { label: "USD", description: "US Dollar" };
