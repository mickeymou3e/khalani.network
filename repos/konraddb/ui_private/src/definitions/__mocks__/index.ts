export * from "./account.mock";
export * from "./assets.mock";
export * from "./balances.mock";
export * from "./orders.mock";
export * from "./rates.mock";
export * from "./store.mock";
export * from "./wallet.mock";
export * from "./tokens.mock";
