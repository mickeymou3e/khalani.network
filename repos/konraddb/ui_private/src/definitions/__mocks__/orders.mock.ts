export const marketSellOrderMock = {
  action: "Sell",
  amount: "1",
  client_code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "98e5e508-453e-44ff-b524-4a451d60cad4",
  created: "2023-07-05T14:29:26.046Z",
  creator_code: "0680f2b1-df2e-48ce-a988-7c7a2d37d8f6",
  customer_code: "1468baf5-b170-4dc0-a7d0-490981abc940",
  executed: {
    amount: "1",
    bank_fee: "0",
    broker_fee: "0",
    price: "1750.85",
    value: "1750.85",
  },
  external_code: "1688567366095579572",
  finished: "2023-07-05T14:29:26.744Z",
  kind: "Regular",
  limit_price: "0",
  market_item: {
    base: "ETH",
    quote: "EUR",
  },
  status: "Filled",
  type: "RFQ_Market",
  user: "",
  role: "",
  userCode: "",
};

export const marketSellOrderMock2 = {
  action: "Sell",
  amount: "1",
  client_code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "c2aba092-e22f-41b5-8d3b-bcf65f92a668",
  created: "2023-07-05T14:18:39.662Z",
  creator_code: "0680f2b1-df2e-48ce-a988-7c7a2d37d8f6",
  customer_code: "1468baf5-b170-4dc0-a7d0-490981abc940",
  executed: {
    amount: "1",
    bank_fee: "0",
    broker_fee: "0",
    price: "1751.49",
    value: "1751.49",
  },
  external_code: "1688566719870229571",
  finished: "2023-07-05T14:18:40.742Z",
  kind: "Regular",
  limit_price: "0",
  market_item: {
    base: "ETH",
    quote: "EUR",
  },
  status: "Filled",
  type: "RFQ_Market",
  user: "",
  role: "",
  userCode: "",
};

export const marketBuyOrderMock = {
  action: "Buy",
  amount: "0.001",
  client_code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "a4df41df-f438-47b9-9219-7d5e9c70eeda",
  created: "2023-05-25T14:17:29.455Z",
  creator_code: "d7ba9aa7-e327-4092-ac90-78f166a8489c",
  customer_code: "1468baf5-b170-4dc0-a7d0-490981abc940",
  executed: {
    amount: "0",
    bank_fee: "0",
    broker_fee: "0",
    partial: true,
    price: "1676.38",
    value: "0",
  },
  finished: "2023-05-25T14:17:29.517Z",
  kind: "Regular",
  limit_price: "0",
  market_item: {
    base: "ETH",
    quote: "EUR",
  },
  status: "Rejected",
  type: "RFQ_Market",
  user: "",
  role: "",
  userCode: "",
};

export const limitBuyOrderMock = {
  action: "Buy",
  amount: "0.01",
  client_code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "739b82b2-0f20-4fdd-95b2-ae73bed4b136",
  created: "2023-05-16T14:57:11.930Z",
  creator_code: "d7ba9aa7-e327-4092-ac90-78f166a8489c",
  customer_code: "1468baf5-b170-4dc0-a7d0-490981abc940",
  executed: {
    amount: "0",
    bank_fee: "0",
    broker_fee: "0",
    partial: true,
    price: "25500.85",
    value: "0",
  },
  external_code: "739b82b2-0f20-4fdd-95b2-ae73bed4b136",
  finished: "2023-05-16T14:57:11.930Z",
  kind: "Partial",
  limit_price: "1",
  market_item: {
    base: "BTC",
    quote: "EUR",
  },
  status: "Canceled",
  type: "RFQ_Limit",
  user: "",
  role: "",
  userCode: "",
};

export const limitSellOrderMock = {
  action: "Sell",
  amount: "1",
  client_code: "a4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "a8e5e508-453e-44ff-b524-4a451d60cad4",
  created: "2023-07-05T12:29:26.046Z",
  creator_code: "a680f2b1-df2e-48ce-a988-7c7a2d37d8f6",
  customer_code: "a468baf5-b170-4dc0-a7d0-490981abc940",
  executed: {
    amount: "1",
    bank_fee: "0",
    broker_fee: "0",
    price: "28510.85",
    value: "28250.85",
  },
  external_code: "2688567366095579572",
  finished: "2023-07-05T12:29:26.744Z",
  kind: "Regular",
  limit_price: "0",
  market_item: {
    base: "BTC",
    quote: "EUR",
  },
  status: "Filled",
  type: "RFQ_Limit",
  user: "",
  role: "",
  userCode: "",
};

export const openLimitBuyOrderMock = {
  action: "Buy",
  amount: "11",
  client_code: "b4137ae0-e941-45b2-a9c3-9f1c16f49428",
  code: "b0bed3c2-54a9-46cd-8a42-0705258d2494",
  created: "2023-11-16T16:33:44.869Z",
  creator_code: "cd5d6c87-b221-4494-92c1-8e6bd5f0fa34",
  customer_code: "26a17540-2bf0-428d-b902-33f035bcbeb7",
  executed: {
    amount: "0",
    bank_fee: "0",
    broker_fee: "0",
    partial: true,
    price: "0",
    value: "0",
  },
  external_code: "b0bed3c2-54a9-46cd-8a42-0705258d2494",
  finished: null,
  kind: "Partial",
  limit_price: "1",
  market_item: {
    base: "JLT-F23",
    quote: "EUR",
  },
  status: "Open",
  type: "RFQ_Limit",
  user: "",
  role: "",
  userCode: "",
};

export const ordersMock = [
  marketSellOrderMock,
  limitBuyOrderMock,
  marketBuyOrderMock,
  marketSellOrderMock2,
  limitSellOrderMock,
  openLimitBuyOrderMock,
];
