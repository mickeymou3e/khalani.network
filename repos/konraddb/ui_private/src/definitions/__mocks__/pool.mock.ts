export const jasminePoolDepositsMock = [
  {
    id: "02e74265-e8f0-4e7a-b826-5c95eb57d509",
    balance: 933,
    tokenId:
      "9099846275235338337781713235466366617387933382847318544154661189209135513600",
    token: {
      vintage: "2023-03-01T00:00:00.000Z",
      generator: {
        techType: "SOLAR",
      },
    },
  },
  {
    id: "07685f05-e427-4e14-ae8c-0f2d80d81d17",
    balance: 901,
    tokenId:
      "58143434673104402184681770351867146451825489264921167162932316293061232558080",
    token: {
      vintage: "2023-02-01T00:00:00.000Z",
      generator: {
        techType: "SOLAR",
      },
    },
  },
  {
    id: "cbff1eca-f4bd-4586-8a0c-99295946cd8f",
    balance: 4600,
    tokenId:
      "97807749870362540913077634561303509279322211818769535094137228738296505434112",
    token: {
      vintage: "2023-02-01T00:00:00.000Z",
      generator: {
        techType: "WIND",
      },
    },
  },
  {
    id: "e24c00d7-adde-469f-9e02-0594a1fb8e4c",
    balance: 700,
    tokenId:
      "3162076973941875473479057173965512843390177904055220565933811656470258778112",
    token: {
      vintage: "2023-01-01T00:00:00.000Z",
      generator: {
        techType: "WIND",
      },
    },
  },
];

const tokenMetadataAttributes = [
  {
    trait_type: "Tech Type",
    value: "SOLAR",
  },
  {
    trait_type: "Generator",
    value: "PR-TEP IVE-1",
  },
  {
    trait_type: "Location",
    value: "PR",
  },
  {
    trait_type: "Certification Type",
    value: "REC",
  },
  {
    trait_type: "Vintage",
    value: "Apr 2023",
  },
  {
    trait_type: "Registry",
    value: "NAR",
  },
];

export const jasmineTokenMetadataMock = [
  {
    id: "e89011c4-1b86-434a-93b9-8e802ea60f4e",
    balance: 933,
    attributes: [...tokenMetadataAttributes],
    tokenId:
      "9099846275235338337781713235466366617387933382847318544154661189209135513600",
  },
  {
    id: "e99011c4-1b86-434a-93b9-8e802ea60f7e",
    balance: 901,
    attributes: [...tokenMetadataAttributes],
    tokenId:
      "58143434673104402184681770351867146451825489264921167162932316293061232558080",
  },
  {
    id: "e10011c4-1b86-434a-93b9-8e802ea60f8e",
    balance: 4600,
    attributes: [...tokenMetadataAttributes],
    tokenId:
      "97807749870362540913077634561303509279322211818769535094137228738296505434112",
  },
  {
    id: "e11011c4-1b86-434a-93b9-8e802ea60f9e",
    balance: 700,
    attributes: [...tokenMetadataAttributes],
    tokenId:
      "3162076973941875473479057173965512843390177904055220565933811656470258778112",
  },
];
