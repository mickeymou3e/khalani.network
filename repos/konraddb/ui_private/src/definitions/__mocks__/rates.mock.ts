export const btcRateMock = {
  bid: {
    price: "27994.49",
    volume: "1",
  },
  ask: {
    price: "28092.65",
    volume: "1",
  },
  timestamp: 1688593185820,
};
export const ethRateMock = {
  bid: {
    price: "1756.58",
    volume: "2.5",
  },
  ask: {
    price: "1761.87",
    volume: "2.5",
  },
  timestamp: 1688593185916,
};
export const atomRateMock = {
  bid: {
    price: "8.590444975",
    volume: "100",
  },
  ask: {
    price: "8.616255025",
    volume: "100",
  },
  timestamp: 1688593187317,
};
export const jltRateMock = {
  bid: {
    price: "2.02",
    volume: "1",
  },
  ask: {
    price: "2.04",
    volume: "1",
  },
  timestamp: 1688593185820,
};

export const ratesMock = {
  btc: btcRateMock,
  eth: ethRateMock,
  atom: atomRateMock,
  jlt_f23: jltRateMock,
};

export const jltWithMarketPriceMock = {
  askPrice: "2.05",
  askPriceUnformatted: "2.045639172207791",
  askPriceValue: 2.045639172207791,
  base: "JLT-F23",
  bidPrice: "2.04",
  bidPriceUnformatted: "2.0354364830695975",
  bidPriceValue: 2.0354364830695975,
  fullName: "JLT-F23",
  id: "JLT-F23/EUR",
  marketPrice: "2.04",
  marketPriceValue: 2.040537827638694,
  pair: "JLT-F23/EUR",
  quote: "EUR",
  selected: false,
  timestamp: "1700140829255",
};
