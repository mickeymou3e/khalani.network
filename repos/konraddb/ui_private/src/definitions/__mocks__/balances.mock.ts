import { customerCodeMock } from "@/definitions/__mocks__/account.mock";
import { AccountType } from "@/definitions/config";

export const balancesMock = [
  {
    accounts: [],
    available: "1000",
    code: "USD",
    is_fiat: true,
    locked: "0",
    name: "US Dollar",
    total: "1000",
  },
  {
    accounts: [],
    available: "7000",
    code: "EUR",
    is_fiat: true,
    locked: "0",
    name: "Euro",
    total: "7000",
  },
  {
    accounts: [],
    available: "25",
    code: "BTC",
    locked: "0",
    name: "Bitcoin",
    total: "25",
  },
  {
    accounts: [],
    available: "0",
    code: "LTC",
    locked: "0",
    name: "Litecoin",
    total: "0",
  },
  {
    accounts: [],
    available: "120",
    code: "ETH",
    locked: "0",
    name: "Ethereum",
    total: "120",
  },
  {
    accounts: [],
    available: "3000",
    code: "ATOM",
    locked: "0",
    name: "Cosmos",
    total: "120",
  },
];

export const tradedBalancesMock = [
  {
    accounts: [
      {
        available: "1000.25",
        code: customerCodeMock,
        locked: "10",
        total: "1010.25",
        type: AccountType.CustomerTrading,
      },
    ],
    available: "1000.25",
    code: "EUR",
    is_fiat: true,
    locked: "10",
    name: "Euro",
    total: "1010.25",
  },
  {
    accounts: [
      {
        available: "50",
        code: customerCodeMock,
        locked: "0",
        total: "50",
        type: AccountType.CustomerTrading,
      },
      {
        available: "0",
        code: "8714bbf0-a2db-443f-aa80-e0025a7b2105",
        locked: "0",
        total: "0",
        type: AccountType.TradingWallet,
      },
      {
        available: "200",
        code: "ef148c21-0799-4b68-930c-11fdcb0ec29d",
        locked: "0",
        total: "200",
        type: AccountType.TradingWallet,
      },
    ],
    available: "250",
    code: "JLT-F23",
    locked: "0",
    name: "JLT-F23",
    total: "250",
  },
];
