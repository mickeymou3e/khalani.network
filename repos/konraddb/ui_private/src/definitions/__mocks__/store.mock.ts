import React from "react";

import { AssetName } from "@/definitions/config";
import {
  AccountPageTabs,
  Backdrops,
  EcoAssets,
  ExecutionSide,
  FiatCurrencies,
  ModalVariants,
  NotificationPropsVariant,
  Notifications,
  OrderType,
  RequestStatusProps,
  SnackbarVariant,
  WalletPageTabs,
  WalletPorfolioTabs,
} from "@/definitions/types";
import { HoldingsActionTypes } from "@/store/ui/ui.types";

export const createJasmineMockApiInitialState = ({
  tokenMetadata = [],
  poolDeposits = [],
}: {
  tokenMetadata?: unknown[];
  poolDeposits?: unknown[];
}): any => ({
  queries: {
    "getTokenMetadata(undefined)": {
      data: tokenMetadata,
    },
    "getPoolDeposits(undefined)": {
      data: poolDeposits,
    },
  },
});

export const createMockApiInitialState = ({
  userProfile = {},
  assignedAssets = [],
  orders = [],
  rates = {},
  balances = [],
  apiTokens = [],
  customer = {},
  custodyWallets = [],
}: {
  userProfile?: unknown;
  assignedAssets?: unknown[];
  rates?: {
    BTC?: object;
    ETH?: object;
    ATOM?: object;
    "JLT-F23"?: object;
  };
  orders?: unknown[];
  balances?: unknown[];
  apiTokens?: unknown[];
  customer?: unknown;
  custodyWallets?: unknown[];
}): any => ({
  queries: {
    "getUserProfile(undefined)": {
      data: userProfile,
    },
    getAssignedAssets: {
      data: assignedAssets,
    },
    'ratesChannel("BTC/EUR")': {
      data: rates?.BTC,
    },
    'ratesChannel("ETH/EUR")': {
      data: rates?.ETH,
    },
    'ratesChannel("ATOM/EUR")': {
      data: rates?.ATOM,
    },
    'ratesChannel("JLT-F23/EUR")': {
      data: rates?.["JLT-F23"],
    },
    "getOrders(undefined)": {
      data: {
        records: orders,
      },
    },
    getBalances: {
      data: balances,
    },
    "getApiTokens(undefined)": {
      data: {
        records: apiTokens,
      },
    },
    "getCryptoDepositsAddressess('b4137ae0-e941-45b2-a9c3-9f1c16f49428')": {
      data: custodyWallets,
    },
    getCustomer: {
      data: customer,
    },
  },
});

export const createMockUiInitialState = ({
  base = AssetName["JLT-F23"],
  quote = FiatCurrencies.EUR,
  selectedHolding = HoldingsActionTypes.Orders,
  hideZeroBalances = false,
  searchText = "",
  pageSize = 10,
  orderType = OrderType.MARKET,
  side = ExecutionSide.BUY,
  hideValues = false,
  wallet = {
    hideValues: false,
    pageSize: 10,
    activeTab: WalletPageTabs.portfolio,
    activePortfolioTab: WalletPorfolioTabs.pool,
  },
  account = {
    hideValues: false,
    pageSize: 10,
    activeTab: AccountPageTabs.profile,
  },
  auxilliary = {
    pageSize: 10,
    ecoAssetType: EcoAssets.Recs,
  },
  modal = {
    variant: null,
    params: null,
  },
  offline = false,
}: {
  base?: string;
  quote?: string;
  selectedHolding?: HoldingsActionTypes;
  hideZeroBalances?: boolean;
  searchText?: string;
  pageSize?: number;
  orderType?: OrderType;
  side?: ExecutionSide;
  hideValues?: boolean;
  wallet?: {
    hideValues?: boolean;
    pageSize?: number;
    activeTab?: WalletPageTabs;
    activePortfolioTab?: WalletPorfolioTabs;
  };
  account?: {
    hideValues?: boolean;
    pageSize?: number;
    activeTab?: AccountPageTabs;
  };
  auxilliary?: {
    pageSize?: number;
    ecoAssetType: EcoAssets;
  };
  modal?: {
    variant?: ModalVariants | string | null;
    params?: any | null;
  };
  offline?: boolean;
}) => ({
  trade: {
    base,
    quote,
    holdings: {
      selected: selectedHolding,
      hideZeroBalances,
      searchText,
      pageSize,
    },
    ticket: {
      orderType,
      side,
    },
  },
  wallet: {
    hideValues: wallet.hideValues ?? hideValues,
    pageSize: wallet.pageSize ?? pageSize,
    activeTab: WalletPageTabs.portfolio,
    activePortfolioTab: wallet.activePortfolioTab ?? WalletPorfolioTabs.pool,
  },
  account: {
    hideValues: account.hideValues ?? hideValues,
    pageSize: account.pageSize ?? pageSize,
    activeTab: AccountPageTabs.profile,
  },
  auxilliary: {
    pageSize: auxilliary.pageSize ?? pageSize,
    ecoAssetType: auxilliary.ecoAssetType,
  },
  modal: {
    variant: modal.variant ?? null,
    params: modal.params ?? null,
  },
  offline,
});

export const createMockNotificationsStore = ({
  id = 0,
  primaryText = "",
  secondaryText = "",
  fixedId = 0,
  text = "",
  buttonText = "",
  variant = SnackbarVariant.success,
  notifications = [],
  link = "",
  error = false,
}: {
  id?: number;
  primaryText?: string;
  secondaryText?: string;
  fixedId?: number;
  text?: string;
  buttonText?: string;
  variant?: SnackbarVariant;
  notifications?: {
    id: Notifications;
    primaryText: string;
    variant: NotificationPropsVariant;
    customChildren: React.ReactNode;
  }[];
  link?: string;
  error?: boolean;
}) => ({
  snackbar: {
    id,
    primaryText,
    secondaryText,
    variant,
    link,
  },
  fixedSnackbar: {
    buttonText,
    id: fixedId,
    text,
    error,
  },
  notifications: notifications.length ? notifications : [],
});

export const createMockBackdropsStore = ({
  currentBackdrop = null,
  parameters = null,
}: {
  currentBackdrop?: Backdrops | null;
  parameters?: any;
}) => ({
  currentBackdrop,
  parameters,
});

export const createMockAuthStore = ({
  csrfToken = "",
  loginToken = "",
  wsToken = "",
  errorCode = 0,
  isFullyLoggedIn = false,
}: {
  csrfToken?: string;
  loginToken?: string;
  wsToken?: string;
  errorCode?: number;
  isFullyLoggedIn?: boolean;
}): any => ({
  csrfToken,
  loginToken,
  wsToken,
  errorCode,
  isFullyLoggedIn,
});

export const createMockWalletStore = ({
  cryptoDepositAddresses = [],
}: {
  cryptoDepositAddresses?: unknown[];
}): any => ({
  cryptoDepositAddresses,
  fetchCryptoDepositAddressesStatus: RequestStatusProps.IDLE,
  depositsHistory: [],
  selectedAsset: "",
  selectedAssetDetails: null,
  cryptoWithdrawalAddresses: [],
  withdrawalsHistory: [],
  requestStatus: RequestStatusProps.IDLE,
});

export const createMockDispatch = () => (): any => ({
  unsubscribe: () => {},
});
