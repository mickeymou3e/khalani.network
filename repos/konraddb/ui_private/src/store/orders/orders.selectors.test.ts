import { ordersMock } from "@/definitions/__mocks__";
import {
  selectSettledOrder,
  selectSettledOrders,
} from "@/store/orders/orders.selectors";

describe("Orders selectors", () => {
  test("should select settled orders", () => {
    const [first, second, third, fourth] =
      selectSettledOrders.resultFunc(ordersMock);

    expect(first).toMatchSnapshot();
    expect(second).toMatchSnapshot();
    expect(third).toMatchSnapshot();
    expect(fourth).toMatchSnapshot();
  });

  test("should select a specific settled order", () => {
    const settledOrders = selectSettledOrders.resultFunc(ordersMock);
    const result = selectSettledOrder(
      "739b82b2-0f20-4fdd-95b2-ae73bed4b136-2023-05-16T14:57:11.930Z"
    ).resultFunc(settledOrders);

    expect(result).toMatchSnapshot();
  });
});
