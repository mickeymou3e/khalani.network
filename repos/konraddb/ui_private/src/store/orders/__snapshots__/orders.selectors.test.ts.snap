// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders selectors should select a specific settled order 1`] = `
{
  "amount": "0.01000",
  "base": "BTC",
  "createdDate": "2023-05-16",
  "createdTime": "14:57:11",
  "date": "2023-05-16",
  "executedAmount": "0.00000",
  "fee": "-",
  "filledPercentage": "",
  "filledPercentageValue": 0,
  "finishedDate": "2023-05-16",
  "finishedTime": "14:57:11",
  "id": "739b82b2-0f20-4fdd-95b2-ae73bed4b136-2023-05-16T14:57:11.930Z",
  "initialAmount": "0.01000",
  "isOpen": false,
  "isPartial": false,
  "orderId": "739b82b2-0f20-4fdd-95b2-ae73bed4b136",
  "pair": "BTC/EUR",
  "price": "25,500.85 EUR",
  "priceBase": "25,500.85",
  "quote": "EUR",
  "role": "",
  "side": "Buy",
  "status": "Canceled",
  "time": "14:57:11",
  "total": "0.01 EUR",
  "type": "RFQ_Limit",
  "user": "",
  "userCode": "",
}
`;

exports[`Orders selectors should select settled orders 1`] = `
{
  "amount": "1.0000",
  "base": "ETH",
  "createdDate": "2023-07-05",
  "createdTime": "14:29:26",
  "date": "2023-07-05",
  "executedAmount": "1.0000",
  "fee": "-",
  "filledPercentage": "",
  "filledPercentageValue": 100,
  "finishedDate": "2023-07-05",
  "finishedTime": "14:29:26",
  "id": "98e5e508-453e-44ff-b524-4a451d60cad4-2023-07-05T14:29:26.046Z",
  "initialAmount": "1.0000",
  "isOpen": false,
  "isPartial": false,
  "orderId": "98e5e508-453e-44ff-b524-4a451d60cad4",
  "pair": "ETH/EUR",
  "price": "1,750.85 EUR",
  "priceBase": "1,750.85",
  "quote": "EUR",
  "role": "",
  "side": "Sell",
  "status": "Filled",
  "time": "14:29:26",
  "total": "1,750.85 EUR",
  "type": "RFQ_Market",
  "user": "",
  "userCode": "",
}
`;

exports[`Orders selectors should select settled orders 2`] = `
{
  "amount": "0.01000",
  "base": "BTC",
  "createdDate": "2023-05-16",
  "createdTime": "14:57:11",
  "date": "2023-05-16",
  "executedAmount": "0.00000",
  "fee": "-",
  "filledPercentage": "",
  "filledPercentageValue": 0,
  "finishedDate": "2023-05-16",
  "finishedTime": "14:57:11",
  "id": "739b82b2-0f20-4fdd-95b2-ae73bed4b136-2023-05-16T14:57:11.930Z",
  "initialAmount": "0.01000",
  "isOpen": false,
  "isPartial": false,
  "orderId": "739b82b2-0f20-4fdd-95b2-ae73bed4b136",
  "pair": "BTC/EUR",
  "price": "25,500.85 EUR",
  "priceBase": "25,500.85",
  "quote": "EUR",
  "role": "",
  "side": "Buy",
  "status": "Canceled",
  "time": "14:57:11",
  "total": "0.01 EUR",
  "type": "RFQ_Limit",
  "user": "",
  "userCode": "",
}
`;

exports[`Orders selectors should select settled orders 3`] = `
{
  "amount": "0.0010",
  "base": "ETH",
  "createdDate": "2023-05-25",
  "createdTime": "14:17:29",
  "date": "2023-05-25",
  "executedAmount": "0.0000",
  "fee": "-",
  "filledPercentage": "",
  "filledPercentageValue": 0,
  "finishedDate": "2023-05-25",
  "finishedTime": "14:17:29",
  "id": "a4df41df-f438-47b9-9219-7d5e9c70eeda-2023-05-25T14:17:29.455Z",
  "initialAmount": "0.0010",
  "isOpen": false,
  "isPartial": false,
  "orderId": "a4df41df-f438-47b9-9219-7d5e9c70eeda",
  "pair": "ETH/EUR",
  "price": "1,676.38 EUR",
  "priceBase": "1,676.38",
  "quote": "EUR",
  "role": "",
  "side": "Buy",
  "status": "Rejected",
  "time": "14:17:29",
  "total": "0.00 EUR",
  "type": "RFQ_Market",
  "user": "",
  "userCode": "",
}
`;

exports[`Orders selectors should select settled orders 4`] = `
{
  "amount": "1.0000",
  "base": "ETH",
  "createdDate": "2023-07-05",
  "createdTime": "14:18:39",
  "date": "2023-07-05",
  "executedAmount": "1.0000",
  "fee": "-",
  "filledPercentage": "",
  "filledPercentageValue": 100,
  "finishedDate": "2023-07-05",
  "finishedTime": "14:18:40",
  "id": "c2aba092-e22f-41b5-8d3b-bcf65f92a668-2023-07-05T14:18:39.662Z",
  "initialAmount": "1.0000",
  "isOpen": false,
  "isPartial": false,
  "orderId": "c2aba092-e22f-41b5-8d3b-bcf65f92a668",
  "pair": "ETH/EUR",
  "price": "1,751.49 EUR",
  "priceBase": "1,751.49",
  "quote": "EUR",
  "role": "",
  "side": "Sell",
  "status": "Filled",
  "time": "14:18:40",
  "total": "1,751.49 EUR",
  "type": "RFQ_Market",
  "user": "",
  "userCode": "",
}
`;
